includes/class-newspack-popups-model.php

```diff
@@ -279,6 +279,115 @@ protected static function create_popup_object( $post, $include_categories = fals
 		return $popup;
 	}
 
+	/**
+	 * Get the popup dismissal text.
+	 *
+	 * @param object $popup The popup object.
+	 * @return string|null Dismiss popup text.
+	 */
+	protected static function get_dismiss_text( $popup ) {
+		return ! empty( $popup['options']['dismiss_text'] ) && strlen( trim( $popup['options']['dismiss_text'] ) ) > 0 ? $popup['options']['dismiss_text'] : null;
+	}
+
+	/**
+	 * Insert analytics tracking code.
+	 *
+	 * @param object $popup The popup object.
+	 * @param string $element_id The id of the popup element.
+	 * @return string Prints the generated amp-analytics element.
+	 */
+	protected static function insert_event_tracking( $popup, $element_id ) {
+		if ( class_exists( '\Google\Site_Kit\Context', '\Google\Site_Kit\Modules\Analytics' ) ) {
+			$analytics           = new \Google\Site_Kit\Modules\Analytics( new \Google\Site_Kit\Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+			$google_analytics_id = $analytics->get_settings()->get()['propertyID'];
+		} else {
+			return '';
+		}
+
+		$event_category = 'Newspack Popup';
+		$event_label    = 'Newspack Popup: ' . $popup['title'];
+
+		$has_link                = preg_match( '/<a\s/', $popup['body'] ) !== 0;
```

Comment:
Do we actually need all of this logic? Would there be any harm in including all the triggers every time, and if the matching selector is absent it simply won't every trigger?

Reply:
Yes, errors are thrown in the JS console if an event handler is attached to a non-existent element :/